File: codeforces/cf1312C_adding_powers.py
# unihernandez22
# https://codeforces.com/contest/1312/problem/C
# greedy, math

t = int(input())
for _ in range(t):
    n, k = map(int, input().split())
    a = list(map(int, input().split()))

    ans = 0
    while any(a):
        ans |= [x%k for x in a if x%k] > [1]
        a = [x//k for x in a]
    
    print("YNEOS"[ans::2])

File: codeforces/cf1312C_adding_powers.rs
// uninhm
// https://codeforces.com/contest/1312/problem/C
// greedy, math

use std::io::stdin;

fn main() {
    let mut t = String::new();
    stdin().read_line(&mut t).unwrap();
    let t = t.trim().parse().unwrap();

    for _ in 0..t {
        let mut line = String::new();
        stdin().read_line(&mut line).unwrap();

        let mut words: Vec<i64> = line
            .split_whitespace()
            .map(|x| x.parse().unwrap())
            .collect();
        
        let n = words[0];
        let k = words[1];
        
        line = String::new();
        stdin().read_line(&mut line).unwrap();

        let mut a: Vec<i64> = line
            .split_whitespace()
            .map(|x| x.parse().unwrap())
            .collect();

        let mut ans = "YES";
        while a.iter().any(|&x| x != 0) {
            let b = a.iter().map(|x| x%k).filter(|&x| x != 0).collect::<Vec<i64>>();
            if b > vec![1] {
                ans = "NO";
                break;
            }
            a = a.iter().map(|x| x/k as i64).collect();
        }
        println!("{}", ans);
    }
}
